### PR TITLE
feat: clan languages

### PR DESCRIPTION
**Pull request by or2e**
**Patch notes:**

## Webui
- Language has been added to clans to make it easier to find a clan.  @or2e 
  - The clan leader can select multiple languages. 
  - Added a list of popular languages. If you do not find your language, please [write here](https://discord.com/channels/279063743839862805/1088203928497303715).
  <details>
    <summary>Screenshots</summary>

    image
    image
    image
    image
  </details>

### CODE REVIEW

File: src/Persistence/Migrations/20240104122127_AddClanLanguages.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
```
Comment (namidaka) [inline src/Persistence/Migrations/20240104122127_AddClanLanguages.cs:4]: to remove

File: src/Persistence/Migrations/20240104122127_AddClanLanguages.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+namespace Crpg.Persistence.Migrations
+{
```
Comment (namidaka) [inline src/Persistence/Migrations/20240104122127_AddClanLanguages.cs:6]: Scope Namespace

File: src/Persistence/Migrations/20240104122127_AddClanLanguages.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AlterDatabase()
+                .Annotation("Npgsql:Enum:activity_log_type", "user_created,user_deleted,user_renamed,user_rewarded,item_bought,item_sold,item_broke,item_reforged,item_repaired,item_upgraded,character_created,character_deleted,character_rating_reset,character_respecialized,character_retired,character_rewarded,server_joined,chat_message_sent,team_hit,clan_armory_add_item,clan_armory_remove_item,clan_armory_return_item,clan_armory_borrow_item")
+                .Annotation("Npgsql:Enum:battle_fighter_application_status", "pending,declined,accepted")
+                .Annotation("Npgsql:Enum:battle_mercenary_application_status", "pending,declined,accepted")
+                .Annotation("Npgsql:Enum:battle_phase", "preparation,hiring,scheduled,live,end")
+                .Annotation("Npgsql:Enum:battle_side", "attacker,defender")
+                .Annotation("Npgsql:Enum:character_class", "peasant,infantry,shock_infantry,skirmisher,crossbowman,archer,cavalry,mounted_archer")
```
Comment (namidaka) [inline src/Persistence/Migrations/20240104122127_AddClanLanguages.cs:19]: Remove Annotation that are not linked to language

File: src/Persistence/Migrations/20240104122127_AddClanLanguages.Designer.cs [hunk 4]
```diff
@@ -0,0 +1,2183 @@
...
+using Crpg.Persistence;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Migrations;
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
+using NetTopologySuite.Geometries;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Metadata;
+
+#nullable disable
+
```
Comment (namidaka) [inline src/Persistence/Migrations/20240104122127_AddClanLanguages.Designer.cs:22]: remove

File: src/Persistence/Migrations/20240104122127_AddClanLanguages.Designer.cs [hunk 5]
```diff
@@ -0,0 +1,2183 @@
...
+namespace Crpg.Persistence.Migrations
+{
```
Comment (namidaka) [inline src/Persistence/Migrations/20240104122127_AddClanLanguages.Designer.cs:24]: scoped namespace

**Review by namidaka — CHANGES_REQUESTED** [review]

File: src/Persistence/Migrations/20240104122127_AddClanLanguages.cs (hunk 1 above)
Comment (or2e) [reply src/Persistence/Migrations/20240104122127_AddClanLanguages.cs]: ++

File: src/Persistence/Migrations/20240104122127_AddClanLanguages.cs (hunk 2 above)
Comment (or2e) [reply src/Persistence/Migrations/20240104122127_AddClanLanguages.cs]: ++

File: src/Persistence/Migrations/20240104122127_AddClanLanguages.cs (hunk 3 above)
Comment (or2e) [reply src/Persistence/Migrations/20240104122127_AddClanLanguages.cs]: ++
I will review and edit automatic migrations in the future

File: src/Persistence/Migrations/20240104122127_AddClanLanguages.Designer.cs (hunk 4 above)
Comment (or2e) [reply src/Persistence/Migrations/20240104122127_AddClanLanguages.Designer.cs]: ++

File: src/Persistence/Migrations/20240104122127_AddClanLanguages.Designer.cs (hunk 5 above)
Comment (or2e) [reply src/Persistence/Migrations/20240104122127_AddClanLanguages.Designer.cs]: ++

**Review by namidaka — APPROVED** [review]

**Pull request merged**